pkg/build/controller/factory/factory.go

```diff
@@ -14,16 +14,17 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
+	builder "github.com/openshift/origin/pkg/build/builder"
 	controller "github.com/openshift/origin/pkg/build/controller"
 	strategy "github.com/openshift/origin/pkg/build/controller/strategy"
-	buildutil "github.com/openshift/origin/pkg/build/util"
 	osclient "github.com/openshift/origin/pkg/client"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
 type BuildControllerFactory struct {
-	Client              *osclient.Client
-	KubeClient          *kclient.Client
+	OSClient            osclient.Interface
+	KubeClient          kclient.Interface
+	BuildClient         builder.BuildClientInterface
```

Comment:
Something like https://github.com/openshift/origin/blob/master/pkg/deploy/generator/config_generator.go#L33 is a good example of a multi-source yet mockable interface - Client is a "easy" implementation.  Or similar https://github.com/openshift/origin/blob/master/pkg/deploy/controller/factory/factory.go#L305

Both accomplish what you're trying to do here, but they avoid the "everyone has to mock methods they don't need" problem.